utmap-server/src/app/main/service/buildingService.py

```diff
@@ -2,116 +2,182 @@
-            'message': 'Found all buildings',
-            'body' : responseBody
+            'status': 'failure',
+            'message': 'Internal server error'
             }
-        return jsonify({'result' : responseObject, 'status' : 200})
+        status = 500
+    return responseObject, status
 
 def getOneBuilding(_id):
-    building = Building()
-    buildings = building.connectToBuildings()
-    build = building.findBuildById(_id, buildings)
-    if build:
-        responseBody = building.formatOneBuild(build)
-        responseObject = {
-        'status': 'success',
-        'message': 'Found one building',
-        'body' : responseBody
-        }
-        return jsonify({'result' : responseObject, 'status' : 200})
-    else:
+    try:
+        building = Building()
+        buildings = building.connectToBuildings()
+        build = building.findBuildById(_id, buildings)
+
+        if build:
+            responseBody = building.formatOneBuild(build)
+            responseObject = {
+                'status': 'success',
+                'message': 'Found one building',
+                'body' : responseBody
+                }
+            status = 200
+        else:
+            responseObject = {
+                'status': 'failure',
+                'message': 'Building not found'
+                }
+            status = 404
+    except(Exception):
         responseObject = {
-        'status': 'fail',
-        'message': 'Building not found'
-        }
-        return jsonify({'result' : responseObject, 'status' : 404})
+            'status': 'failure',
+            'message': 'Internal server error'
+            }
+        status = 500
+    return responseObject, status
 
 def deleteOneBuilding(_id):
-    building = Building()
-    buildings = building.connectToBuildings()
-    if building.deleteBuildById(_id, buildings):
-        responseObject = {
-            'status': 'success',
-            'message': 'Building successfully deleted'
-        }
-        return jsonify({'result' : responseObject, 'status' : 200})
-    else:
+    try:
+        building = Building()
+        buildings = building.connectToBuildings()
+
+        if building.deleteBuildById(_id, buildings):
+            responseObject = {
+                'status': 'success',
+                'message': 'Building successfully deleted'
+                }
+            status = 200
+        else:
+            responseObject = {
+                'status': 'failure',
+                'message': 'Building not found'
+                }
+            status = 404
+    except(Exception):
         responseObject = {
-            'status': 'fail',
-            'message': 'Building not found'
-        }
-        return jsonify({'result' : responseObject, 'status' : 404})
+            'status': 'failure',
+            'message': 'Internal server error'
+            }
+        status = 500
+    return responseObject, status
     
+def countKeys(data):
```

Comment:
It may be better to rename `countKeys` and `isValidInput` to be more specific, eg. `countBuildingKeys`. Same for the functions in `eventService.py`

Reply:
Can do